src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -16,6 +19,78 @@ public void setup() {
 
     @Test
     public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+        assertFalse(productDao.findProducts().isEmpty());
+    }
+
+    @Test
+    public void testSaveNewProduct() {
+        Currency usd = Currency.getInstance("USD");
+        Product product = new Product("test-product", "Samsung Galaxy S", new BigDecimal(100), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
+        productDao.save(product);
+
+        assertTrue(product.getId() > 0);
+        try {
+            Product result = productDao.getProduct(product.getId());
+            assertNotNull(result);
+            assertEquals("test-product", result.getCode());
+        } catch (ProductNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testSaveModifiedProduct() {
+        Currency usd = Currency.getInstance("USD");
+        Product product = new Product("test-product", "Samsung Galaxy S", new BigDecimal(100), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
+        productDao.save(product);
+
+        try {
+            Product modifiedProduct = productDao.getProduct(product.getId());
+            modifiedProduct.setCode("test-product1");
+            Long oldId = product.getId();
+            productDao.save(modifiedProduct);
+            modifiedProduct = productDao.getProduct(product.getId());
+            assertEquals(oldId, modifiedProduct.getId());
+            assertNotEquals(null, "test-product", modifiedProduct.getCode());
+        } catch (ProductNotFoundException e) {
+            e.printStackTrace();
```

Comment:
It's considered as bad practice to simply printStackTrace on exception. Please put at least comprehensible message when you handle exception like this in test.